M2/Macaulay2/e/m2tbb.hpp

```diff
@@ -4,6 +4,11 @@
 // The plan: All uses of TBB go through the following interface.
 
 #include <M2/config.h>                    // make sure WITH_TBB is set before including mtbb.hpp
+
+#ifndef WITH_TBB
+#define MATHICGB_NO_TBB 1
+#endif
```

Comment:
Should this be in `mathicgb/mtbb.hpp` instead?

Reply:
I don't think so.  If we still want mathicgb to exist as a standalone library (I maintain it as one in Debian/Ubuntu, and I think Singular has the option of linking against it), then I don't think we should be putting Macaulay2 macros in it.  (`WITH_TBB` is M2's and `MATHICGB_NO_TBB` is mathicgb's)

Reply:
`WITH_TBB` is already in there ...

Reply:
Ah indeed it is!  Here's the current mathicgb behavior:

```c++
#ifndef WITH_TBB
#ifndef MATHICGB_NO_TBB
#define WITH_TBB 1
#endif
#endif

#ifdef WITH_TBB
// do tbb stuff
```
The current behavior seems reasonable -- if neither macro is defined, then default to using TBB.  That makes sense since if you're building mathicgb by itself, it will define `MATHICGB_NO_TBB` in the build flags when you call `./configure --without-tbb` but won't define `WITH_TBB`.  I'm proposing that we mimic that by defining `MATHICGB_NO_TBB` before we include the header.

Reply:
I'm confused. Why even set `MATHICGB_NO_TBB`? It's not used anywhere if `WITH_TBB` is already set. I think this is complicating the logic unnecessarily.

Reply:
Ahh I see what's happening. I think this is a bug just in autotools, because CMake defines `MATHICGB_NO_TBB` in compile arguments.